pyproject.toml

```diff
@@ -0,0 +1,159 @@
+[project]
+name        = "pact-python"
+description = "Tool for creating and verifying consumer-driven contracts using the Pact framework."
+dynamic     = ["version"]
+
+authors     = [{ name = "Matthew Balvanz", email = "matthew.balvanz@workiva.com" }]
+maintainers = [{ name = "Joshua Ellis", email = "josh@jpellis.me" }]
+
+readme = "README.md"
+license = { file = "LICENSE" }
+keywords = ["pact", "contract-testing", "testing"]
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Console",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: MIT License",
+  "Operating System :: MacOS :: MacOS X",
+  "Operating System :: Microsoft :: Windows",
+  "Operating System :: POSIX :: Linux",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3 :: Only",
+  "Topic :: Software Development :: Testing",
+]
+
+requires-python = ">=3.8,<4.0"
+dependencies = [
+  "click      ~= 8.1",
+  "fastapi    ~= 0.103",
+  "psutil     ~= 5.9",
+  "requests   ~= 2.31",
+  "six        ~= 1.16",
+  "uvicorn    ~= 0.13",
+]
+
+[project.urls]
+"Homepage"      = "https://pact.io"
+"Repository"    = "https://github.com/pact-foundation-pact-python"
+"Documentation" = "https://docs.pact.io"
+"Bug Tracker"   = "https://github.com/pact-foundation/pact-python/issues"
+"Changelog"     = "https://github.com/pact-foundation/pact-python/blob/master/CHANGELOG.md"
+
+[project.scripts]
+pact-verifier = "pact.cli.verify:main"
+
+[project.optional-dependencies]
+types = ["mypy ~= 1.1", "types-requests ~= 2.31"]
+test = [
+  "coverage[toml] ~= 7.3",
+  "httpx          ~= 0.24",
+  "mock           ~= 5.1",
+  "pytest         ~= 7.4",
+  "pytest-cov     ~= 4.1",
+  "testcontainers ~= 3.7",
+  "yarl           ~= 1.9",
```

Comment:
How do I install these optional-deps

I've already ran 

```console
hatch run example 
ImportError while loading conftest '/Users/yousaf.nabi/saf_stuff/dev/pact-foundation/pact-python/examples/conftest.py'.
examples/conftest.py:15: in <module>
    from yarl import URL
E   ModuleNotFoundError: No module named 'yarl'
```

managed to get pip to install with `pip install -e '.[test]'` 

but still get the same error as above

Reply:
Hatch should installed these dependencies automatically when it first creates the virtual env. When dependencies change, hatch _should_ automatically detect this and (un)install the changed dependencies; but I have encountered this issue myself once. When I re-created the venv though, all dependencies were correctly installed.

I suspect this is an edge case whereby hatch does not detect transitive dependencies. In particular, there is a chain `pact-python[dev] -> pact-python[test] -> yarl` and `pact-python[dev]`, and the default venv should install all `[dev]` dependencies:

https://github.com/pact-foundation/pact-python/blob/7279cac38ce15e15bb95ef364ea14f829d2e6e07/pyproject.toml#L101-L103

I'll check whether this is a known issue upstream.

Reply:
Thank you! I think I was a bit early to the party for a review, but good to see how its going to play out!

Reply:
ahh okay, so its happier now i've done the following  (been able to find yarl and at least run the examples)

```
git clean -fdx
hatch env remove
hatch env create
```